internal/storage/storage.go

```diff
@@ -1,17 +1,56 @@
 package storage
 
 import (
+	"io"
+	"log"
 	"sync"
+
+	"github.com/sirupsen/logrus"
 )
 
+type Config struct {
+	StoragePath string
+}
+
 type Storage struct {
-	sync.Mutex // mutex for lock
-	storage    map[string]string
+	sync.Mutex  // mutex for lock
+	storagePath string
+	storage     map[string]string
 }
 
-func New() *Storage {
-	return &Storage{
-		storage: make(map[string]string),
+func New(cfg Config) *Storage {
```

Comment:
```suggestion
func New(cfg Config) (*Storage, error) {
```
никаких вызовов log.Fatal внутри пакетов. Данный вызов допустим только в main. Если посмотреть внутрь log.Fatal, то там дергается os.Exit(1), который обрывает все defer во всем приложении, принудительно останавливает все горутины и по сути роняет приложение. По хорошему любая ошибка, возникшая на любом уровне вложенности должна подниматься вверх (main, http-handler и т.д.) и там наверху должно приниматься решение об обработке данной ошибки.